### PR TITLE
[log-shipper] fix Elasticsearch destination default values

### PR DESCRIPTION
**Pull request by yalosev**
## Description
Fix ClusterLogDestination default values for Elasticsearch
Close https://github.com/deckhouse/deckhouse/issues/474

## Why do we need it, and what problem does it solve?
Elasticsearch destination depends on Elasticsearch version. Current default values break default configuration on most often Elasticsearch installations.

## Changelog entries

```changes
module: log-shipper
type: fix
description: Fix default CRD values
note: ClusterLogDestination created in v1.29.0-alpha should be recreated
```

### CODE REVIEW

File: modules/460-log-shipper/docs/USAGE.md [hunk 1]
```diff
@@ -192,7 +193,7 @@ spec:
   type: Elasticsearch
   elasticsearch:
     endpoint: http://192.168.1.1:9200
-    docTypeSupport: true
+    docType: "vector" # !!!
```
Comment (nabokihms) [inline modules/460-log-shipper/docs/USAGE.md:196]: What does the symbol `!!!` mean?

File: modules/460-log-shipper/crds/cluster-log-destination.yaml [hunk 2]
```diff
@@ -220,18 +220,19 @@ spec:
...
                       description: |
                         Use for storage indexes or datastreams (https://www.elastic.co/guide/en/elasticsearch/reference/master/data-streams.html).
 
-                        Datastream usage is better as logs and metrics storage.
-                    docTypeSupport:
-                      type: boolean
-                      default: false
+                        Datastream usage is better for logs and metrics storage but they works only for Elasticsearch >= 7.16.X.
+                    docType:
+                      type: string
```
Comment (nabokihms) [inline modules/460-log-shipper/crds/cluster-log-destination.yaml:229]: Could you elaborate more on why do we need string type here? As for me, enum with three options seems more preferable.

File: modules/460-log-shipper/docs/USAGE.md (hunk 1 above)
Comment (yalosev) [reply modules/460-log-shipper/docs/USAGE.md:196]: pay attention to this line :)

File: modules/460-log-shipper/crds/cluster-log-destination.yaml (hunk 2 above)
Comment (yalosev) [reply modules/460-log-shipper/crds/cluster-log-destination.yaml:229]: Users can have their own index created and configured with their custom doc_type. We can't predict every value for the doc_type. Also, this field is needed only by 5.X ES

File: modules/460-log-shipper/docs/USAGE.md (hunk 1 above)
Comment (yalosev) [reply modules/460-log-shipper/docs/USAGE.md]: changed

File: modules/460-log-shipper/crds/cluster-log-destination.yaml (hunk 2 above)
Comment (nabokihms) [reply modules/460-log-shipper/crds/cluster-log-destination.yaml:229]: My concern is if I have an instance of 5.x ElasticSearch, how do I know which docType should I set? Can we provide some commands to find it? If not, it is ok.

**Review by nabokihms — DISMISSED** [review]
Approved, but I advise waiting for the @konstantin-axenov final review.

File: modules/460-log-shipper/crds/cluster-log-destination.yaml [hunk 3]
```diff
@@ -220,18 +220,19 @@ spec:
...
+                      default: false
                       description: |
                         Use for storage indexes or datastreams (https://www.elastic.co/guide/en/elasticsearch/reference/master/data-streams.html).
 
-                        Datastream usage is better as logs and metrics storage.
-                    docTypeSupport:
-                      type: boolean
-                      default: false
+                        Datastream usage is better for logs and metrics storage but they works only for Elasticsearch >= 7.16.X.
+                    docType:
```
Comment (konstantin-axenov) [inline modules/460-log-shipper/crds/cluster-log-destination.yaml:228]: I can't see changes in ru doc

File: modules/460-log-shipper/crds/cluster-log-destination.yaml [hunk 4]
```diff
@@ -220,18 +220,19 @@ spec:
...
+                      type: string
                       description: |
                         The doc_type for your index data. This is only relevant for Elasticsearch <= 6.X.
 
-                        You do not need this option if you are using Elasticsearch >= 7.0 since this version has removed doc_type mapping.
+                        For ES >= 7.X - You do not need this option since this version has removed doc_type mapping
+                        For ES >= 6.X - preferred value is a '_doc', we this value will be easier to upgrade to 7.X
```
Comment (konstantin-axenov) [inline modules/460-log-shipper/crds/cluster-log-destination.yaml:234]:
```suggestion
                        For ES >= 6.X - preferred value is a '_doc', this value will be easier to upgrade to 7.X
```

File: modules/460-log-shipper/crds/cluster-log-destination.yaml (hunk 2 above)
Comment (yalosev) [reply modules/460-log-shipper/crds/cluster-log-destination.yaml:229]: If you (as a client) don't mind. You can set 'vector' value. I left it as an example in README and CRD.
If you have some special case (for example you have created an index with your own doc_type, then I think you know, what you have set here).

File: modules/460-log-shipper/crds/cluster-log-destination.yaml (hunk 3 above)
Comment (yalosev) [reply modules/460-log-shipper/crds/cluster-log-destination.yaml:228]: Indeed. Fixed

**Review by konstantin-axenov — APPROVED** [review]

**Pull request merged**